extensions/guacamole-auth-sso/modules/guacamole-auth-sso-base/src/main/java/org/apache/guacamole/auth/sso/user/SSOAuthenticatedUser.java

```diff
@@ -47,58 +47,51 @@
     private Credentials credentials;
     
     /**
-     * Tokens associated with this authenticated user.
+     * The groups that this user belongs to.
      */
-    private Map<String, String> tokens;
-
+    private Set<String> effectiveGroups;
+    
     /**
-     * The unique identifiers of all user groups which this user is a member of.
+     * Parameter tokens to be automatically injected for any connections used
+     * by this user.
      */
-    private Set<String> effectiveGroups;
+    private Map<String, String> tokens;
 
     /**
-     * Initializes this AuthenticatedUser using the given username and
-     * credentials, and an empty map of parameter tokens.
+     * Initializes this SSOAuthenticatedUser, associating it with the given
+     * username, credentials, groups, and parameter tokens. This function must
+     * be invoked for every SSOAuthenticatedUser created.
      *
      * @param username
      *     The username of the user that was authenticated.
      *
      * @param credentials
      *     The credentials provided when this user was authenticated.
-     */
-    public void init(String username, Credentials credentials) {
-        this.init(username, credentials, Collections.emptyMap(), Collections.emptySet());
-    }
-    
-    /**
-     * Initializes this AuthenticatedUser using the given username,
-     * credentials, and parameter tokens.
      *
-     * @param username
-     *     The username of the user that was authenticated.
+     * @param effectiveGroups
+     *     The groups that the authenticated user belongs to.
      *
-     * @param credentials
-     *     The credentials provided when this user was authenticated.
-     * 
      * @param tokens
      *     A map of all the name/value pairs that should be available
-     *     as tokens when connections are established with this user.
+     *     as tokens when connections are established by this user.
      */
     public void init(String username, Credentials credentials,
-            Map<String, String> tokens, Set<String> effectiveGroups) {
+            Set<String> effectiveGroups, Map<String, String> tokens) {
         this.credentials = credentials;
-        this.tokens = Collections.unmodifiableMap(tokens);
```

Comment:
Why get rid of these conversions here? Like wrapping the tokens in an unmodifable map, or lowercaseing the username?

Reply:
I think this is mostly just git deciding that this file is the most similar to the CAS version of things, not that these have been truly removed, but I'll double-check.

Reply:
OK - I've restored the unmodifiable map and lowercase transformation. The latter was indeed accidentally removed from CAS.